src/event/value.rs

```diff
@@ -89,8 +89,10 @@ impl From<BTreeMap<String, Value>> for Value {
     }
 }
 
-impl From<Vec<Value>> for Value {
-    fn from(value: Vec<Value>) -> Self {
+impl<T: Into<Value>> From<Vec<T>> for Value {
+    fn from(value: Vec<T>) -> Self {
+        let value = value.into_iter().map(Into::into).collect::<Vec<_>>();
+
```

Comment:
I can't see why this change was necessary?

Reply:
Ah yeah, I had a different implementation at first that resulted in a `Vec<String>`, which I wanted to be able to call `Value::from` on.

I can revert this, although this does seem like a more flesible implementation, but I'm not sure if there might be any performance hit, I'd have to see the resulting assembly to know for sure.

Reply:
Would it not allocate a new vector for it?

Reply:
Ah yeah, I had a different implementation at first that resulted in a `Vec<String>`, which I wanted to be able to call `Value::from` on.

I can revert this, although this does seem like a more flexible implementation, but I'm not sure if there might be any performance hit, I'd have to see the resulting assembly to know for sure.

(technically, `Vec<Value>` should result in a no-op conversion, but I'm not sure if the compiler can optimize it down to a single `Value::Array(value)` call)

Reply:
That's what the assembly could teach us. Technically the input and output of the iterator are the same if `T` is a `Value`, and the iterator isn't mutating state, so it _could_ optimize it away, but I'm not sure if it does.

Reply:
Looking at https://godbolt.org/z/qccK5q, it does look like there is more going on under the hood with this impl, even if you pass in a `Vec<Value>`, but I'm by no means an expert on Assembly or optimizing Rust code.

I'd be curious to hear @bruceg's thoughts though

Reply:
You need to add the flag `-C opt-level=3` to see the optimised output, but that does obscure what it is doing even further..

Reply:
Ah, yes, of course. There's still a difference in the compiled output, so it's not 1:1 equal, but it does reduce the amount of overhead.

Regardless, we can back out of this, as I'm not using it right now, but this was an interesting exercise 😄 and I'm still happy to hear Bruce's thoughts.

Reply:
Indeed. Nothing better than digging through assembler! It's a shame the compiler doesn't allow both implementations. 

I think I would prefer the old implementation, unless it can be proven to have no overhead. It is a very simple one line for any code that does need to covert the vector into a vector of Values, and the compiler does make sure you can't error here. Although I don't imagine this would be called in a tight loop, so the overhead is likely to be incredibly minimal..

Reply:
Yeah, I'll back this out, since it's not used any more, regardless 👍

Reply:
I would be surprised if it can optimize out the creation of a new vector and drop the old, there's just too much going on with memory allocations. The question is if it matters for performance, and it looks like you eliminated that path anyways.

Reply:
This came up while discussing #4066 !